tests/ui/fn/fn-ptr-eq.rs

```diff
@@ -0,0 +1,19 @@
+// check-pass
+
+fn foo(_: &()) {}
+
+fn main() {
+    let x: for<'a> fn(&'a ()) = foo;
+    let y: for<'a> fn(&'a ()) = foo;
+    bar(x, y);
+
+    let x: fn(&'static ()) = foo;
+    let y: for<'a> fn(&'a ()) = foo;
+    x == y;
```

Comment:
So it works if the RHS is higher-ranked, but not if the LHS is higher-ranked? That is odd...

Reply:
that seems definitely fixable, but I think it's a preexisting issue of binary ops.

Reply:
I assume this is related to this remark?

> == has too much magic to work with this.

I don't I understand what the magic is, I thought `a == b` was just `PartialEq::eq(&a, &b)`...